src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,135 @@
+            String line = in.nextLine();
+            String[] splits = line.split(" \\| ");
+            initializeTask(splits);
+        }
+    }
+
+    private void initializeTask(String[] splits) throws DukeException {
+        switch (splits[0]) {
+        case "T":
+            storedTaskList.addTodo(splits[2]);
+            break;
+        case "D":
+            storedTaskList.addDeadline(splits[2], splits[3]);
+            break;
+        case "E":
+            storedTaskList.addEvent(splits[2], splits[3]);
+            break;
+        default:
+            break;
+        }
+        if (splits[1].equals("1")) {
+            storedTaskList.markTask(storedTaskList.getSize());
+        }
+    }
+    // End of task initialization
+
+    public void rewriteDukeFile(TaskList taskList) throws DukeException {
+        try {
+            if (!Files.exists(Paths.get(filePath))) {
+                File dukeFile = new File(filePath);
+                dukeFile.createNewFile();
+            }
+
+            FileWriter dukeFileWriter = new FileWriter(filePath, false);
+            for (Task task : taskList.getTasks()) {
+                String output = retrieveTaskInformationForFileStorage(task);
+                dukeFileWriter.append(output);
+            }
+            dukeFileWriter.close();
+        } catch (IOException exception) {
+            throw new StorageOutputException();
+        }
+    }
+
+    public void appendDukeFile(TaskList taskList) throws DukeException {
+        try {
+            FileWriter dukeFileWriter;
+            if (Files.exists(Paths.get(filePath))) {
+                dukeFileWriter = new FileWriter(filePath, true);
+            } else {
+                File dukeFile = new File(filePath);
+                dukeFile.createNewFile();
+                dukeFileWriter = new FileWriter(filePath, false);
+            }
+
+            Task newTask = taskList.getTasks().get(taskList.getTasks().size() - 1);
+            String output = retrieveTaskInformationForFileStorage(newTask);
+            dukeFileWriter.append(output);
+            dukeFileWriter.close();
+        } catch (IOException exception) {
+            throw new StorageOutputException();
+        }
+    }
+
+    private static String retrieveTaskInformationForFileStorage(Task task) {
+        String output = "";
+        if (task instanceof Todo) {
+            output = String.format("%s | %s | %s", "T", task.isDone() ? "1" : "0", task.getTaskName());
+        }
+        if (task instanceof Deadline) {
+            output = String.format("%s | %s | %s | %s", "D", task.isDone() ? "1" : "0",
+                    task.getTaskName(), ((Deadline) task).getDeadlineTime());
+        }
+        if (task instanceof Event) {
+            output = String.format("%s | %s | %s | %s", "E", task.isDone() ? "1" : "0",
+                    task.getTaskName(), ((Event) task).getEventTime());
+        }
+        output += System.lineSeparator();
+        return output;
```

Comment:
Each type of Task should know how to print itself, i.e., it should have a `toString` method that `retrieveTaskInformationForFileStorage` calls, rather than `retrieveTaskInformationForFileStorage` itself having code for each task type -- this isn't extensible; if you add a new task type, you would need to change this method, which is strange.

Reply:
Thanks for the advice! I didn't use toString() method as it may be confused with the print format used for normal listing messages. However, the idea of polymorphism is still utilized to resolve the issue: 

In Task.java
```
public String getTaskInStorageFormat() {
        String format = String.format("%s | %s",this.isDone() ? "1" : "0", this.getTaskName());
        return format;
}
```
In Todo.java
```
@Override
public String getTaskInStorageFormat() {
    String format = String.format("%s | %s", "T", super.getTaskInStorageFormat());
    return format;
}
```
In Deadline.java (similar to event)
```
@Override
public String getTaskInStorageFormat() {
    String format = String.format("%s | %s | %s", "D", super.getTaskInStorageFormat(), this.getDeadlineTime());
    return format;
}
```

In Storage.java
```
private static String retrieveTaskInformationForFileStorage(Task task) {
    String output = task.getTaskInStorageFormat();
    output += System.lineSeparator();
    return output;
}
```